### ISSUE TITLE
Change how the OOD evaluation frame is processed for STARK verification

### ISSUE THREAD

**Issue by Al-Kindi-0** [0xPolygonMiden/miden-vm#875]
Currently in Winterfell, the trace OOD (Out-of-Domain) evaluation frame is composed of $2$ rows, representing the current and next rows,

$$
[T_0(z),\cdots, T_N(z)]
$$

and

$$
[T_0(gz),\cdots, T_N(gz)]
$$

where $z$ is the OOD element, $g$ is the trace domain generator and $N:= 72 + 9 - 1$ is the total number of trace columns minus 1.

The way they are currently being read is done row-wise meaning that $[T_0(z),\cdots, T_N(z)]$ is read first and then $[T_0(gz),\cdots, T_N(gz)]$ is read next. This means that when we read the OOD frame from the advice provider into memory, it will be laid out in a way that is difficult to work with as we would like to have, at each memory location, the word $[T_i(z)_0, T_i(z)_1, T_i(gz)_0, T_i(gz)_1]$ available where $T_i(z) = (T_i(z)_0, T_i(z)_1)$ and $T_i(gz) = (T_i(gz)_0, T_i(gz)_1)$ when working over a quadratic extension field.
A proposal to fix this is to change the way the Winterfell Prover sends the OOD frame to the Verifier to be  

$$
[T_0(z), T_0(gz),\cdots, T_N(z), T_N(gz)]
$$

In other words, the OOD frame will be laid out in memory, word by word, as:

```math
[[ T_0(z)_0, T_0(z)_1, T_0(gz)_0, T_0(gz)_1 ], \cdots [ T_{71}(z)_0, T_{71}(z)_1, T_{71}(gz)_0, T_{71}(gz)_1 ], 
```

```math
[ T_{72 + 0}(z)_0, T_{72 + 0}(z)_1, T_{72 + 0}(gz)_0, T_{72 + 0}(gz)_1 ], \cdots [ T_{72+8}(z)_0, T_{72+8}(z)_1, T_{72+8}(gz)_0, T_{72+8}(gz)_1 ]]
```

This will also mean that instead of reseeding the random coin twice as we currently do, we will have to reseed only once.

**Comment by Al-Kindi-0** [comment]
This has now being implemented in Winterfell
